File: week2-math/problem01-팩토리얼/heetae.java
/**
문제: 팩토리얼
접근 방식: 순차로 순회해서 넘어갈경우 i를 반환 어짜피 n이 10!까지임
시간복잡도: O(10)
*/
class Solution {
    public int solution(int n) {
        int answer = 1;
        for(int i=0;i<10;i++){
            answer*=(i+1);
            if(answer>n) return i;
        }
        return 10;
    }
}

File: week1-array-string/problem09-시저암호/heetae.py
# 문제: 시저 암호
# 접근 방식: 공백은 그대로 두고, 각 영문의 코드에 n을 더한 후 대문자와 소문자 범위를 넘으면 알파벳 처음으로 돌린다.
# 시간복잡도: O(n) (n은 s의 길이)

def solution(s, n):
    answer = list(s)
    for i in range(0, len(answer)):
        if answer[i] == ' ':
            continue

        current = ord(answer[i]) + n

        if ord(answer[i]) >= 65 and ord(answer[i]) <= 90:
            if current > 90:
                current = current - 26
            answer[i] = chr(current)
        else:
            if current > 122:
                current = current - 26
            answer[i] = chr(current)

    return ''.join(answer)
